fem/conduitdatacollection.cpp

```diff
@@ -774,15 +774,16 @@ ConduitDataCollection::MeshToBlueprintMesh(Mesh *mesh,
       n_bndry_topo["type"]     = "unstructured";
       n_bndry_topo["coordset"] = coordset_name;
 
-      Element::Type bndry_ele_type = mesh->GetBdrElementType(0);
-
-      std::string bndry_ele_shape = ElementTypeToShapeName(bndry_ele_type);
+      int num_bndry_ele = mesh->GetNBE();
 
+      // must initialize this to something
+      Element::Type bndry_ele_type   = (num_bndry_ele > 0) ? mesh->GetBdrElementType(
+                                          0) : mfem::Element::POINT;
+      std::string bndry_ele_shape    = ElementTypeToShapeName(bndry_ele_type);
       n_bndry_topo["elements/shape"] = bndry_ele_shape;
 
-
-      int num_bndry_ele = mesh->GetNBE();
-      int bndry_geom    = mesh->GetBdrElementBaseGeometry(0);
+      int bndry_geom          = (num_bndry_ele > 0) ? mesh->GetBdrElementBaseGeometry(
+                                   0) : mfem::Element::POINT;
```

Comment:
I'm concerned that arbitrarily choosing `Element::POINT` might be problematic in some cases.

Possible alternatives for the empty boundary case on a given rank:
* Use an `MPI_Allreduce` to find the value
* deduce the boundary element type based on the volumetric mesh elements ? E.g. if the volume mesh is `Element::HEXAHEDRON`, the boundary type should be `Element::QUADRILATERAL`.

My concern is about the case where rank 0 has no boundary elements, but the incorrect value is used for the blueprint index. VisIt might interpret the overall boundary elements incorrectly.

This might also be a useful utility function to promote to the `DataCollection` base class.

Reply:
@kennyweiss the blueprint index does not have any details about the element types, however it will provide overall topology type metadata (like `unstructured` vs `rectilinear`). I recommend trying to see if there are problems (and even if there are -- we can adapt to mitigate them)

Reply:
(I am pretty sure this wont trip things up, but we should check to avoid surprises )

Reply:
I originally had an `MPI_Allreduce` but I was concerned about that for a few reasons:  1) it's more expensive, although this is a startup cost, 2) what if it's different among ranks? what's the right one to pick?, 3) the rest of the code sort of assumes the value is the same everywhere, 4) it's never actually used as far as I can tell, 5) I did what was done in `MeshBlueprintToMesh`, which also makes this assumption.  I'm happy making an assumption based on the volumetric type.  I'm also happy putting useful functions in the base class.